File: Selnium script for testing/Script with Python/test_scenario_16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

def execute_test_16():
    """
    Dummy test run for scenario 16
    """
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    driver.get("https://dummy.example.com/page_16")

    driver.find_element(By.XPATH, "//button[@type='submit']").click()
    driver.find_element(By.ID, "checkout_btn").click()
    time.sleep(2)
    driver.find_element(By.NAME, "password_field").send_keys("test_data_717")
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.NAME, "password_field").send_keys("test_data_530")
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.CSS_SELECTOR, ".promo-code").send_keys("test_data_382")
    time.sleep(2)
    driver.find_element(By.CLASS_NAME, "nav-item").click()
    driver.find_element(By.ID, "checkout_btn").click()
    time.sleep(2)

    driver.quit()

if __name__ == "__main__":
    execute_test_16()

File: Selnium script for testing/Script with Java/src/test/java/testscenario/TestScenario16.java
package testscenario;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import java.time.Duration;

public class TestScenario16 {
    public static void executeTest16() {
        /*
         * Dummy test run for scenario 16
         */
        WebDriver driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get("https://dummy.example.com/page_16");

        try {
            driver.findElement(By.className("nav-item")).sendKeys("test_data_643");
            if (!driver.findElement(By.cssSelector(".promo-code")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.id("user_name")).sendKeys("test_data_850");
            if (!driver.findElement(By.name("password_field")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            if (!driver.findElement(By.name("password_field")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.name("password_field")).sendKeys("test_data_595");
            driver.findElement(By.id("checkout_btn")).click();
            if (!driver.findElement(By.linkText("Log Out")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.linkText("Log Out")).sendKeys("test_data_296");
            if (!driver.findElement(By.linkText("Log Out")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.name("password_field")).sendKeys("test_data_793");
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
    
    public static void main(String[] args) {
        executeTest16();
    }
}
